logic/src/commonMain/kotlin/com/wire/kalium/logic/functional/EitherFlow.kt

```diff
@@ -5,3 +5,12 @@ import kotlinx.coroutines.flow.filter
 import kotlinx.coroutines.flow.map
 
 fun <L, R> Flow<Either<L, R>>.onlyRight(): Flow<R> = filter { it.isRight() }.map { (it as Either.Right<R>).value }
+
+fun <L, R, T> Flow<Either<L, R>>.mapLeft(block: suspend (L) -> T): Flow<Either<T, R>> =
+    map { it.fold({ l -> Either.Left(block(l)) }) { r -> Either.Right(r) } }
+
+fun <L, R, T> Flow<Either<L, R>>.mapRight(block: suspend (R) -> T): Flow<Either<L, T>> =
+    map { it.fold({ l -> Either.Left(l) }) { r -> Either.Right(block(r)) } }
+
+fun <L, R, T : Any> Flow<Either<L, R>>.foldEither(fnL: suspend (L) -> T, fnR: suspend (R) -> T): Flow<T> =
+    map { it.fold({ l -> fnL(l) }, { r -> fnR(r) }) }
```

Comment:
It would be better to write some short explanation in the comment for others to understand how it should behave 😄

Reply:
that one was removed and the others are quit self-explanatory IMO :)